app/grandchallenge/hanging_protocols/forms.py

```diff
@@ -47,7 +47,7 @@ def __init__(self, *args, **kwargs):
             ButtonHolder(Submit("save", "Save")),
         )
 
-    def clean_json(self):
+    def clean_json(self):  # noqa: C901

```

Comment:
Could you split the method out into sub-validation methods? Make the aim of each method clear so the future dev can see what is happening from `clean_json` (which should do say 4 method calls).